internal/env/env.go

```diff
@@ -0,0 +1,68 @@
+// package env contains data structures and methods to assist with manageing environment variables.
+package env
+
+import (
+	"runtime"
+	"strings"
+)
+
+// Env represents a map of environment variables. The keys may be case-sensitive.
+// If they are, the original casing is lost.
+type Env struct {
+	env           map[string]string
+	caseSensitive bool
+}
+
+// New return a new Env. If caseSensitive is true, keys are case-sensitive, otherwise they are not.
```

Comment:
I think `New` should have no args and do the thing that makes the most sense for the current OS, mainly because `env.New(true)` / `env.New(false)` has low information about the meaning of the boolean so you have to look at docs. 

And therefore the other constructor should take an arg to control case sensitivity, ... or maybe make it a method?

```go
e := env.New()
e.SetCaseSensitive(true)
```

Reply:
Another approach is the "bool enum"...

```go
const (
	CaseInsensitive = false
	CaseSensitive   = true
)
```

Reply:
I like that too, but I found 3 options functions, so I think this is ok.